### PR TITLE
Fix ":262 in error" when running tests_init()

### PR DESCRIPTION
**Pull request by caijun**
As reported in tdhock/animint#66, this branch fixes the `:262 in error` when running `tests_init()` after `tests_init("firefox)`. My idea is to get process ids by port number used in animint. `tests_init()` should run in the environment that no processes occupy the port number. Also, this makes the code more succinct comparing by using `pid.txt`.

The branch works well in creating and exiting remote server of selenium on my Mac OS X 10.10.3, even though the following error and warning occur.

```
Error in startServer(host, port, app) : Failed to create server
Calls: <Anonymous> -> <Anonymous> -> startServer
Execution halted
```
This error is likely to result from `httpuv::startServer()`, which I haven't trace back.
```
Warning message:
running command 'lsof -i :4444 -t' had status 1 
```
This warning has no influence to our functions. Hence, I will suppress it in the future.

@tdhock and @cpsievert can review the changes and give comments.

### CODE REVIEW

**Comment by tdhock** [comment]
I tried running the tests and they paused/stopped during the test-facet-trivial.R tests.

File: R/testHelpers.R [hunk 1]
```diff
@@ -164,3 +160,12 @@ find_test_path <- function(dir = ".") {
...
   file.path(dir, ext_dir)
 }
+
+# get process id via port number
+port2pid <- function(port = 4444) {
+#   args <- paste0("-i :", port, " -t")
+#   pid <- as.integer(system2("lsof", args = args, stdout = TRUE))
+  cmd <- paste0("lsof -i :", port, " -t")
+  pid <- as.integer(system(cmd, intern = TRUE))
+  pid
```
Comment (cpsievert) [inline R/testHelpers.R]: this won't work cross-platform

**Comment by cpsievert** [comment]
I looked into killing process(es) by port number a while ago, but I couldn't find a nice cross-platform way to do that. 

I think you're problems in #66 are due to selenium not shutting down properly (which might be a problem with RSelenium internals) so let's make sure we identify the issue before spending anymore time here.

**Comment by cpsievert** [comment]
I reopen this if we can't sort out #66 without it, but I'm pretty this isn't the source of our problems...

File: R/testHelpers.R (hunk 1 above)
Comment (caijun) [inline R/testHelpers.R]: Yes, it is a solution for Unix-like OS, not Windows.

**Comment by caijun** [comment]
I update #66 , it is a problem of  `RSelenium::closeServer()`, which doesn't kill `java` process properly.

**Pull request closed without merging**